### ISSUE TITLE
Hide Telegram bot token from trace log

### ISSUE THREAD

**Issue by d-Rickyy-b** [Radarr/Radarr#10476]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Is your feature request related to a problem? Please describe

While investigating an issue I am currently facing with Telegram notifications, I noticed that the bot token is not removed in the trace log. For a second I thought about reporting this as a security related issue, but on second thought, trace logs are only shared upon request and hence the risk is limited.

Still, I think that any API key (especially those used for notifications) should be excluded from the log.

Currently it looks like this (don't worry, it's a fake token I made up):

```
2024-09-24 14:20:02.1|Trace|HttpClient|Req: [POST] https://api.telegram.org/bot1234567890:qJwWPXLTawgbkQ_owcxuBEaVEICacLq4/sendmessage: chat_id=123456&parse_mode=HTML&text=<text>
```

### Describe the solution you'd like

For other log entries, the log contains `(removed)` as in this log line:

```
2024-09-24 13:54:33.5|Trace|HttpClient|Res: HTTP/1.1 [GET] http://sabnzbd:8080/api?mode=queue&start=0&limit=0&category=movies&apikey=(removed)&output=json: 200.OK (608 bytes) (3 ms)
```

From my understanding, this is done via the "CleansingRules". The solution to the issue would be to add a new CleansingRule to prevent the Telegram bot token to be logged. The token is part of the URL.
`https://api.telegram.org/bot<token>/<method>`.

Such a rule already exists for Discord:
https://github.com/Radarr/Radarr/blob/a7eb4a4a0454ffab149a34a800abe519eceeaf6a/src/NzbDrone.Common/Instrumentation/CleanseLogMessage.cs#L55-L56

### Describe alternatives you've considered

N/A

### Anything else?

This feature request doesn't only apply to Telegram notifications but to all notification services that could log an api key. Since I am only using Telegram as a notification provider, I can't say which other services this affects.

Also this request obviously is not limited to Radarr but also applies to the other *arrs.

**Comment by bakerboy448** [comment]
Thanks for the report!

**Comment by d-Rickyy-b** [comment]
Thanks for providing a fix. Do you need a separate bug report for the other *arr projects or will this fix be merged into the other projects as well?

**Comment by RobinDadswell** [comment]
> Thanks for providing a fix. Do you need a separate bug report for the other *arr projects or will this fix be merged into the other projects as well?

I have popped a fix in for review in Sonarr (https://github.com/Sonarr/Sonarr/pull/7244) and once that is merged it will open Issues/PRs in our other projects which will then get merged
